janitor/utils.py

```diff
@@ -405,3 +406,219 @@ def _wrapped(x, *args, **kwargs):
             return return_val
 
     return _wrapped
+
+
+def _check_instance(entry: Dict):
+    """
+    Function to check instances in the expand_grid function
+    This checks if entry is a dictionary,
+    checks the instance of value in key:value pairs in entry,
+    and makes changes to other types as deemed necessary. 
+    """
+    # entry has to be a dictionary
+    if not entry:
+        raise ValueError("passed dictionary cannot be empty")
+    # if it is a number or string, wrap in a list
+    entry = {
+        key: [value]
+        if isinstance(value, (type(None), int, float, bool, str))
+        else value
+        for key, value in entry.items()
+    }
+
+    # convert to list if value is a string|tuple|range
```

Comment:
If I am reading the code correctly, this should be:

```suggestion
    # convert to list if value is a set|tuple|range
```